Classes/Queue/RedisQueue.php

```diff
@@ -296,10 +295,22 @@ protected function connectClient()
     {
         $host = isset($this->clientOptions['host']) ? $this->clientOptions['host'] : '127.0.0.1';
         $port = isset($this->clientOptions['port']) ? $this->clientOptions['port'] : 6379;
+        $password = isset($this->clientOptions['password']) ? $this->clientOptions['password'] : '';
         $database = isset($this->clientOptions['database']) ? $this->clientOptions['database'] : 0;
         // The connection read timeout should be higher than the timeout for blocking operations!
         $timeout = isset($this->clientOptions['timeout']) ? $this->clientOptions['timeout'] : round($this->defaultTimeout * 1.5);
-        $connected = $this->client->connect($host, $port, $timeout) && $this->client->select($database);
+
+        $connected = $this->client->connect($host, $port, $timeout);
+        if ($connected) {
+            if ($password !== '') {
+                $authSuccess = $this->client->auth($password);
+                if ($authSuccess !== true) {
+                    throw new JobQueueException('Redis authentication failed.', 1536735535);
+                }
+            }
+            $connected = $this->client->select($database);
```

Comment:
Looking at this change again, I wonder:
Previously `$this->client->select()` was not called if the initial connection failed. Now it is called even if it failed in line 303. Is that maybe a bug?

Reply:
```
$connected = $connected && $this->client->select($database);
```
would reproduce the old behavior.
But probably it should rather throw an exception if the database could not be selected. This could be a separate change of course. just thinking out loud

Reply:
No, the `select()` is in the `if`, so it's only called if `$connected` is true.

Also, mind the reconnection mechanism… Failing auth can fail hard, but anything else could go away… unless you "fix" the wrong password on the Redis side…

Reply:
Yes, I didn't see that in the diff. All good